src/components/dialog/dialog.interactions.stories.tsx

```diff
@@ -0,0 +1,232 @@
+import React, { useRef, useState } from "react";
+import { StoryObj } from "@storybook/react";
+import { userEvent, within } from "@storybook/test";
+
+import Dialog from ".";
+import Button from "../button";
+import userInteractionPause from "../../../.storybook/utils/user-interaction-pause";
+import { allowInteractions } from "../../../.storybook/interaction-toggle/reduced-motion";
+import Typography from "../typography";
+import Textbox from "../textbox";
+import Form from "../form";
+
+type Story = StoryObj<typeof Dialog>;
+
+export default {
+  title: "Dialog/Interactions",
+  component: Dialog,
+  parameters: { layout: "padded" },
+  decorators: [],
+};
+
+export const OpenAndCloseDialog: Story = {
```

Comment:
comment: as this component's open state can be controlled by a prop we don't need this test